src/Service/Resolvers/SqlMutationEngine.cs

```diff
@@ -825,5 +919,37 @@ private HttpContext GetHttpContext()
         {
             return _httpContextAccessor.HttpContext!;
         }
+
+        /// <summary>
+        /// Helper method to construct a transactionscope object at Read Committed isolation level
+        /// with the TransactionScopeAsyncFlowOption option enabled
+        /// </summary>
+        /// <seealso cref="https://learn.microsoft.com/en-us/dotnet/framework/data/transactions/implementing-an-implicit-transaction-using-transaction-scope"/>
+        /// <seealso cref="https://learn.microsoft.com/en-us/dotnet/api/system.transactions.transactionscopeoption?view=net-6.0#fields" />
+        /// <seealso cref="https://learn.microsoft.com/en-us/dotnet/api/system.transactions.transactionscopeasyncflowoption?view=net-6.0#fields" />
+        /// <returns>TransactionScope object set at Read Committed isolation level</returns>
+        private static TransactionScope ConstructReadCommittedTransactionScope()
+        {
+            return new(TransactionScopeOption.Required,
+                           new TransactionOptions
+                           {
+                               IsolationLevel = System.Transactions.IsolationLevel.ReadCommitted
+                           },
+                           TransactionScopeAsyncFlowOption.Enabled);
+        }
+
+        /// <summary>
+        /// Helper method to throw a DataApiBuidlerException when exceptions related
+        /// to transactions are encountered.
+        /// </summary>
+        /// <exception cref="DataApiBuilderException"></exception>
+        private static void ConstructAndThrowDataApiBuilderException()
```

Comment:
nit: since its throwing transaction specific error, the method name should indicate that.